javalib/src/main/scala/java/util/Map.scala

```diff
@@ -27,6 +33,126 @@ trait Map[K, V] {
   def entrySet(): Set[Map.Entry[K, V]]
   def equals(o: Any): Boolean
   def hashCode(): Int
+
+  @JavaDefaultMethod
+  def getOrDefault(key: Any, defaultValue: V): V = {
+    val value = get(key)
+    if (value != null || containsKey(key))
```

Comment:
To avoid looking up the key in the map a *second* time in the extremely common case were `value != null`.

Reply:
Hmm... sounds like a false optimization (IMHO an efficiency oriented implementation is supposed to override this anyways, like `HashMap` in this PR). But if you feel strongly about it, maybe just put a comment? (basically explicitly stating the assumption that `value != null` is the common case and hence this is less lookup on average).

Reply:
I was going to write a comment for this, because I think that optimization is clearly an easy win for the cases where the `key` is in the map. But I realized that we don't know whether the key being in the map is itself the common case. So I reverted this to the dumb version:
```scala
if (containsKey(key)) get(value)
else defaultValue
```